operator/v1/types_ingress.go

```diff
@@ -229,6 +236,138 @@ var (
 	DNSReadyIngressConditionType = "DNSReady"
 )
 
+// SecuritySpec defines the settings for securing IngressController connections.
+type SecuritySpec struct {
+	// profile is one of "Old", "Intermediate", "Modern" or "Custom". "Old",
+	// "Intermediate" and "Modern" profiles map to security configurations from [1]:
+	//
+	// [1] https://wiki.mozilla.org/Security/Server_Side_TLS#Recommended_configurations
+	//
+	// When a profile of type "Old", "Intermediate" or "Modern" is set, the CustomSettings
+	// field is forbidden.
+	//
+	// customSettings must be provided if, and only if, profile is "Custom".
+	//
+	// If unset, the "Intermediate" profile is used.
+	//
+	Profile SecurityProfileType `json:"profile"`
+	// customSettings defines the schema for settings of a "Custom" profile
+	// and is ignored unless a "Custom" profile is specified.
+	//
+	// +optional
+	CustomSettings *CustomProfileSettings `json:"customSettings,omitempty"`
+
+}
+
+// SecurityProfileType defines a security profile.
+type SecurityProfileType string
```

Comment:
For the sake of discussion, imagine for a moment that `CustomProfileSettings` was named `Profile` but was otherwise unchanged.

Does each SecurityProfileType ultimately map _only_ to a certain state of a `Profile`? That is, the type implies _nothing else_ abstracted by the ingress controller or operator?

Maybe another way: given a specific `Profile` and no other input, can you determine the SecurityProfileType?

Reply:
> Maybe another way: given a specific Profile and no other input, can you determine the SecurityProfileType?

Yes, I believe so.